posthog/queries/test/test_trends.py

```diff
@@ -68,5382 +78,5800 @@ def _create_cohort(**kwargs):
-                    Filter(
-                        data={
-                            "date_from": "2019-12-24",
-                            "interval": "hour",
-                            "events": [{"id": "sign up"}],
-                            "breakdown": json.dumps([cohort.pk]),
-                            "breakdown_type": "cohort",
-                        }
-                    ),
-                    self.team,
-                )
-            self.assertEqual(response[0]["labels"][3], "24-Dec-2019 03:00")
-            self.assertEqual(response[0]["data"][3], 1.0)
-            # 217 - 24 - 1
-            self.assertEqual(response[0]["data"][192], 3.0)
-
-            # test week
-            with freeze_time("2020-01-02"):
-                response = trends().run(
-                    Filter(
-                        data={
-                            # 2019-11-24 is a Sunday
-                            "date_from": "2019-11-24",
-                            "interval": "week",
-                            "events": [{"id": "sign up"}],
-                            "breakdown": json.dumps([cohort.pk]),
-                            "breakdown_type": "cohort",
-                        }
-                    ),
-                    self.team,
-                )
+        self.assertEqual(response[0]["count"], 2)
+        self.assertEqual(response[0]["data"][-1], 2)
 
-            self.assertEqual(
-                response[0]["labels"][:5], ["24-Nov-2019", "1-Dec-2019", "8-Dec-2019", "15-Dec-2019", "22-Dec-2019"]
+    def test_bar_chart_by_value(self):
+        self._create_events()
+
+        with freeze_time("2020-01-04T13:00:01Z"):
+            # with self.assertNumQueries(16):
+            response = Trends().run(
+                Filter(
+                    data={
+                        "date_from": "-7d",
+                        "events": [{"id": "sign up"}, {"id": "no events"}],
+                        "display": TRENDS_BAR_VALUE,
+                    }
+                ),
+                self.team,
             )
-            self.assertEqual(response[0]["data"][:5], [0.0, 0.0, 0.0, 0.0, 1.0])
+        self.assertEqual(response[0]["aggregated_value"], 4)
+        self.assertEqual(response[1]["aggregated_value"], 1)
+        self.assertEqual(
+            response[0]["days"],
+            [
+                "2019-12-28",
+                "2019-12-29",
+                "2019-12-30",
+                "2019-12-31",
+                "2020-01-01",
+                "2020-01-02",
+                "2020-01-03",
+                "2020-01-04",
+            ],
+        )
 
-            # test month
-            with freeze_time("2020-01-02"):
-                response = trends().run(
-                    Filter(
-                        data={
-                            "date_from": "2019-9-24",
-                            "interval": "month",
-                            "events": [{"id": "sign u
```

Comment:
Is this assumption correct @neilkakkar Asking because we haven't been running this test before in CI, so it got of sync after #12302

Reply:
Looks like this comment went on the wrong line. :/ This file is huge, super hard to work with in GitHub UI.
What I meant to comment on is the "Zero person IDs shouldn't be counted" explanation I added to `test_weekly_active_users_daily_based_on_action_with_zero_person_ids.

Reply:
Yes, in person on events mode, we discard all events where personID. iszero

https://github.com/PostHog/posthog/blob/master/posthog/queries/trends/trends_event_query.py#L99